File: marinetes/user-mobile/app/src/main/kotlin/com/marinetes/network/serializations/marinetes/user/api/UserAddress.kt
package com.marinetes.network.serializations.marinetes.user.api

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class UserAddress(
  val id: String,
  val category: Int,
  val type: Int,
  val city: String,
  val neighborhood: String,
  val number: String,
  val complement: String? = null,
  val rooms: Int,
  @SerialName("square_meters")
  val squareMeters: Double,
  val state: String,
  val street: String,
  val title: String,
  @SerialName("zip_code")
  val zipCode: String,
)

File: marinetes/monorepo-legacy/packages/database/src/models/UserAddress.ts
import type {
  UserAddress,
  UserAddressCategory,
  UserAddressType,
} from '@marinetes/types/model';
import {
  Entity,
  Column,
  BaseEntity,
  CreateDateColumn,
  UpdateDateColumn,
  PrimaryGeneratedColumn,
} from 'typeorm';

@Entity('user_adresses')
export class UserAddressRepository extends BaseEntity implements UserAddress {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column('varchar')
  title: string;

  @Column('varchar')
  type: UserAddressType;

  @Column('varchar')
  category: UserAddressCategory;

  @Column('int')
  rooms: number;

  @Column('numeric')
  square_meters: number;

  @Column('varchar')
  zip_code: string;

  @Column('varchar')
  state: string;

  @Column('varchar')
  city: string;

  @Column('varchar')
  neighborhood: string;

  @Column('varchar')
  street: string;

  @Column('int')
  number: number;

  @CreateDateColumn()
  created_at: string;

  @UpdateDateColumn()
  updated_at: string;
}
